src/scripts/macros/skill-save-prompt/skill-save-prompt.ts

```diff
@@ -0,0 +1,236 @@
+import { CharacterPF2e } from "@actor";
+import { ActionDefaultOptions } from "@system/action-macros/types.ts";
+import { adjustDC, calculateDC, calculateSimpleDC, DCAdjustment } from "@module/dc.ts";
+import { getActions, loreSkillsFromActiveParty, loreSkillsFromActors } from "./helpers.ts";
+import { ProficiencyRank } from "@item/data/index.ts";
+import { PROFICIENCY_RANKS } from "@module/data.ts";
+import { htmlQuery, signedInteger, tagify } from "@util";
+import * as R from "remeda";
+
+interface SkillSavePromptDialogOptions extends ApplicationOptions {
+    actors: CharacterPF2e[];
+}
+
+interface SkillSavePromptDialogData {
+    proficiencyRanks: SelectData[];
+    dcAdjustments: SelectData[];
+}
+
+interface SelectData {
+    value: string;
+    label: string;
+}
+
+interface TagifyValue {
+    id: string;
+    value: string;
+}
+
+class SkillSavePromptDialog extends Application<SkillSavePromptDialogOptions> {
+    #actions?: Record<string, string>;
+    #lores?: Record<string, string>;
+
+    static override get defaultOptions(): ApplicationOptions {
+        return {
+            ...super.defaultOptions,
+            classes: ["dialog"],
+            id: "generate-skill-save-prompt",
+            template: "systems/pf2e/templates/gm/skill-save-prompt/skill-save-prompt.hbs",
+            title: game.i18n.localize("PF2E.Actor.Party.SkillSavePrompt.Title"),
+            width: 400,
+        };
+    }
+
+    override async getData(): Promise<SkillSavePromptDialogData> {
+        this.#actions = await getActions();
+        this.#lores = this.options.actors ? loreSkillsFromActors(this.options.actors) : loreSkillsFromActiveParty();
```

Comment:
```suggestion
        this.#lores = loreSkillsFromActors(this.options.actors ?? game.actors.party.members ?? []);
```

You can get rid of loreSkillsFromActiveParty by doing this.